### ISSUE TITLE
Error when initializing model using TF1.12

### ISSUE THREAD

**Issue by Hugh0120** [zzh8829/yolov3-tf2#33]
I just wanna try your Keras version model in TF1.12 and added two lines into 'yolov3_tf2/models.py' 

```
if __name__ == '__main__':
    model = YoloV3(training=True, size=418)
```
Things go well in TF2.0, however, when I run it in TF1.12, the following error occurred:

> I just wanna try your Keras version model in TF1.12 and added two lines into 'yolov3_tf2/models.py' 

```
if __name__ == '__main__':
    model = YoloV3(training=True, size=418)
```

However, the following error occurred:

> Traceback (most recent call last):
  File "/Users/xxx/Code/GitOA/xxx/YoloV3/src/yolov3_tf2/models.py", line 312, in <module>
    model = YoloV3(training=True, size=418)
  File "/Users/xxx/Code/GitOA/xxx/YoloV3/src/yolov3_tf2/models.py", line 207, in YoloV3
    x = YoloConv(256, name='yolo_conv_1')((x, x_61))
  File "/Users/xxx/Code/GitOA/xxx/YoloV3/src/yolov3_tf2/models.py", line 112, in yolo_conv
    return Model(inputs, x, name=name)(x_in)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py", line 554, in __call__
    outputs = self.call(inputs, *args, **kwargs)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/keras/engine/network.py", line 815, in call
    mask=masks)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/keras/engine/network.py", line 1002, in _run_internal_graph
    output_tensors = layer.call(computed_tensor, **kwargs)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/keras/layers/convolutional.py", line 194, in call
    outputs = self._convolution_op(inputs, self.kernel)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/nn_ops.py", line 966, in __call__
    return self.conv_op(inp, filter)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/nn_ops.py", line 591, in __call__
    return self.call(inp, filter)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/nn_ops.py", line 208, in __call__
    name=self.name)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/gen_nn_ops.py", line 1026, in conv2d
    data_format=data_format, dilations=dilations, name=name)
  File "/Users/xxx/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 529, in _apply_op_helper
    (input_name, err))
ValueError: Tried to convert 'input' to a tensor and failed. Error: Dimension 1 in both shapes must be equal, but are 13 and 26. Shapes are [?,13,13,512] and [?,26,26,512].
    From merging shape 0 with other shapes. for 'yolo_conv_1/conv2d_59/Conv2D/packed' (op: 'Pack') with input shapes: [?,13,13,512], [?,26,26,512].

I think it's because of the version of Tensorflow but I'm not sure which specific part caused this, does anybody knows?

**Comment by zzh8829** [comment]
maybe there some api difference in tf2 and tf1.12
I never tested this on tf1.12 so can't really answer

**Comment by hollowgalaxy** [comment]
I found same issue in tf 1.13

**Comment by hollowgalaxy** [comment]
corrected [this](https://github.com/zzh8829/yolov3-tf2/blob/e5130786a33dcc612e2617949616defff34237c0/yolov3_tf2/models.py#L112) line and shape issue went away. However this is a terrible duct tape solution, and I am not sure how best to report this for tensorflow team. Here is the code to fix it.
```
return Model(inputs, x, name=name)([x_in[0],x_in[1]] if isinstance(x_in, tuple) else x_in)
```

**Comment by iamsaksham** [comment]
I faced the same issue on mac.
What was happening was that `tensorflow-gpu==2.0.0a0` from the environment.yml was not getting installed as tensorflow-gpu is not supported in mac.
So as a result there was no `tensorflow 2` installed. To fix this I replaced the `tensorflow-gpu==2.0.0a0` with `tensorflow==2.0.0alpha0`

environment.yml
```
name: yolov3-tf2
channels:
  - conda-forge

dependencies:
  - python==3.6
  - pip
  - matplotlib
  - pip:
    - absl-py
    - numpy
    - opencv-python==4.0.0.21
    - tensorflow==2.0.0alpha0
    # - tensorflow-gpu==2.0.0a0
```
